pkg/startupmigrations/migrations.go

```diff
@@ -281,10 +281,12 @@ var backwardCompatibleMigrations = []migrationDescriptor{
 		// Introduced in v20.2.
 		name:   "create new system.tenants table",
 		workFn: createTenantsTable,
-		// NB: no dedicated cluster version was introduced for this table at the
-		// time (4272248e573cbaa4fac436b0ea07195fcd648845). The below is the first
-		// cluster version that was added after the system.tenants table.
-		includedInBootstrap: clusterversion.ByKey(clusterversion.AlterColumnTypeGeneral),
+		// This migration does not have a dedicated cluster version key but was
```

Comment:
I think you get to just remove the workFn here; this migration is "baked in" so to speak.

Reply:
I originally left this in because if I didn't some backup tests would fail, complaining that `system.tenants` doesn't exist. Among other tests, `TestPaginatedBackupTenant`.

 I've looked into this some more and now think I surfaced an unrelated bug. In the bootstrap package, `system.tenants` gets created only if this is the system tenant. However the migration doesn't seem to care about this. Case in point if I change the current diff to add  `if !r.codec.ForSystemTenant() { return nil }` to `createTenantsTable` the test fails.

Reply:
The other tests with problems are `TestBackupRestoreInsideTenant/tenant-backup` and `TestBackupRestoreInsideTenant`, somewhat unsurprisingly.

Reply:
I came up with what I believe to be an acceptable fix in that in `backup_planning.go` I wrap the query to `system.tenants` in `if p.ExecCfg().Codec.ForSystemTenant() { ... }` which should be a no-op, AFAIK the table is empty for non-system tenants. Like I said it probably shouldn't even exist in those cases, yet it does in these backup tests. Is this an artifact of how we set up those test clusters or a real bug in the startupmigrations? I still haven't been able to tell.